proto/asn1-pdu/asn1_pdu_to_der.cc

```diff
@@ -0,0 +1,159 @@
+#include "asn1_pdu_to_der.h"
+
+namespace asn1_pdu {
+
+// The maximum level of recursion allowed. Values greater than this will just
+// fail.
+static constexpr size_t kRecursionLimit = 200;
+
+// Signal that the recursion limit has been exceeded by a
+// message or value and to abort further processing.
+static bool kErrRecursionLimitReached = false;
+
+uint8_t ASN1PDUToDER::GetVariableIntLen(const uint64_t value,
+                                          const size_t base) {
+  uint8_t base_bits = log2(base);
+  for (uint8_t num_bits = (sizeof(value) - 1) * __CHAR_BIT__;
```

Comment:
`__CHAR_BIT__` isn't the same as `CHAR_BIT`. You want the latter, which you get by

`#include <climits>`

See https://en.cppreference.com/w/cpp/types/climits for handy reference :)

(And throughout the file)